python/llm/src/bigdl/llm/transformers/models/utils.py

```diff
@@ -315,10 +315,15 @@ def use_esimd_sdp(q_len, k_len, head_dim, query_states):
     elif query_states.dtype != torch.float16:
         # esimd_sdp only has optimization for FP16 now
         return False
+    elif query_states.shape[0] > 1 and attention_mask is not None:
+        # for batched input, can't accept attention_mask
```

Comment:
For batched input and all zero attention mask, it has no return and will be None?

Reply:
Oh yes... Thanks for pointing that out!
Maybe should change `use_esimd_sdp` into below way ?
```python
def use_esimd_sdp(q_len, k_len, head_dim, query_states):
    if head_dim != 128:
        # esimd_sdp only support head_dim = 128 now
        return False
    elif q_len != 1:
        # esimd_sdp only support rest token now
        return False
    elif k_len < 8:
        # esimd_sdp will cause wrong output when k_len < 8
        return False
    elif query_states.device.type != "xpu":
        # esimd_sdp only support GPU now
        return False
    elif query_states.dtype != torch.float16:
        # esimd_sdp only has optimization for FP16 now
        return False
    elif query_states.shape[0] > 1 and attention_mask is not None:
        # for batched input, can't accept attention_mask
        if len(torch.nonzero(attention_mask, as_tuple=True)) > 1:
            return False

    device_name = torch.xpu.get_device_name(query_states.device.index)
    if device_name.startswith("Intel(R) Arc(TM) A") or \
              device_name.startswith("Intel(R) Data Center GPU Flex") or \
              device_name.startswith("Intel(R) Data Center GPU Max"):
        import linear_fp16_esimd
        if not hasattr(linear_fp16_esimd, "sdp_forward"):
            return False
    else:
        return False

    return True
```

Reply:
Seems good!